Demo/Sources/Views/PlaybackSlider.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  Copyright (c) SRG SSR. All rights reserved.
+//
+//  License information is available from the LICENSE file.
+//
+
+import Player
+import SwiftUI
+
+#if os(iOS)
+struct PlaybackSlider: View {
+    let progressTracker: ProgressTracker
+    let minimumValueText: String?
+    let maximumValueText: String?
+
+    @Binding private var value: Float
+    @State private var valueWidth: CGFloat = 0
+    @State private var previousValueWidth: CGFloat = 0
+
+    @Binding private var buffer: Float
+    @State private var bufferWidth: CGFloat = 0
+
+    @State private var isDragging = false {
+        didSet {
+            progressTracker.isInteracting = isDragging
+        }
+    }
+
+    var body: some View {
+        HStack {
+            text(minimumValueText)
+            progressBar(valueWidth: valueWidth, bufferWidth: bufferWidth)
+            text(maximumValueText)
+        }
+        .frame(maxWidth: .infinity, minHeight: 10, maxHeight: 10)
+    }
+
+    init(progressTracker: ProgressTracker, minimumValueText: String? = nil, maximumValueText: String? = nil) {
+        self.progressTracker = progressTracker
+        self.minimumValueText = minimumValueText
+        self.maximumValueText = maximumValueText
+        _value = Binding(progressTracker, at: \.progress)
+        _buffer = Binding(progressTracker, at: \.loaded)
+    }
+
+    @ViewBuilder
+    private func text(_ text: String?) -> some View {
+        if let text {
+            Text(text)
+                .font(.caption)
+                .monospacedDigit()
+        }
+    }
+
+    @ViewBuilder
+    private func rectangle(opacity: Double = 1, width: CGFloat? = nil) -> some View {
+        Rectangle()
+            .foregroundColor(.white)
+            .opacity(opacity)
+            .frame(width: width)
+    }
+
+    @ViewBuilder
+    private func progressBar(valueWidth: CGFloat, bufferWidth: CGFloat) -> some View {
+        GeometryReader { geometry in
+            ZStack(alignment: .leading) {
+                rectangle(opacity: 0.1)
+                rectangle(opacity: 0.3, width: bufferWidth)
+                rectangle(width: valueWidth)
+            }
+            .gesture(dragGesture(geometry: geometry))
+            .onChange(of: value) { onValueChanged($0, geometry: geometry) }
+            .onChange(of: buffer) { onBufferChanged($0, geometry: geometry) }
+        }
+        .cornerRadius(5)
+        .scaleEffect(y: isDragging ? 1.5 : 1.0)
+        .animation(.smooth(duration: 0.4), value: isDragging)
+    }
+
+    private func dragGesture(geometry: GeometryProxy) -> some Gesture {
+        DragGesture()
+            .onChanged { gesture in
+                isDragging = true
```

Comment:
Maybe it would be even possible to enter interaction mode on touch down (similar to the native player UI experience)?

Reply:
It's a good idea! What do you think about creating an issue for that? 
It may not be easy to handle it very quickly.

Reply:
Sure, the current result is great already.